lib/webauthn/authenticator_assertion_response.rb

```diff
@@ -42,7 +42,7 @@ def initialize(authenticator_data:, signature:, user_handle: nil, **options)
     def verify(expected_challenge, expected_origin = nil, public_key:, sign_count:, user_verification: nil,
                rp_id: nil)
       super(expected_challenge, expected_origin, user_verification: user_verification, rp_id: rp_id)
-      verify_item(:signature, credential_cose_key(public_key))
+      verify_item(:signature, WebAuthn::PublicKey.deserialize(public_key))
```

Comment:
Is there any other part of the gem where we should be calling `WebAuthn::PublicKey.deserialize` that I'm missing?

Reply:
Should this

https://github.com/cedarcode/webauthn-ruby/blob/cdc101c3db9e5d9c0d5c2a3708a086e8f48ca850/lib/webauthn/authenticator_data/attested_credential_data.rb#L55-L59

be

```ruby
WebAuthn::PublicKey.deserialize(public_key).valid?
```
instead?

Reply:
My question was more directed to places where we actually needed to account for the _two formats_.

Answering the question you are now asking: I think it could!
But, I think I'd like to have a deeper discussion on whether all the usages of `COSE::Key.deserialize` should be replaced with `WebAuthn::PublickKey.deserialize` or not.

It's a pretty interesting discussion to have. 
Do we want to make it part of this PR, or discuss separately?
I'm fine either way.

Reply:
> My question was more directed to places where we actually needed to account for the _two formats_.
> 
> Answering the question you are now asking: I think it could!
> But, I think I'd like to have a deeper discussion on whether all the usages of `COSE::Key.deserialize` should be replaced with `WebAuthn::PublickKey.deserialize` or not.

I think in most cases "Yes", given that the WebAuthn spec builds extra domain logic on top of COSE Keys (in https://www.w3.org/TR/webauthn-2/#sctn-attested-credential-data). So ideally I imagine no references to COSE key other than from `WebAuthn::PublicKey`, so that every webauthn specific login is contained there.

> 
> It's a pretty interesting discussion to have.
> Do we want to make it part of this PR, or discuss separately?

Separate PR is fine.
This provides value already in it's current form I think.

> I'm fine either way.

Reply:
Perfect! Let's keep working on this on separate PRs.